model/src/main/java/de/evoila/cf/broker/model/BaseServiceInstanceRequest.java

```diff
@@ -35,11 +36,12 @@
     @JsonProperty("maintenance_info")
     private MaintenanceInfo maintenanceInfo;
 
-    public Map<String, Object> getContext() {
+    public Context getContext() {
         return context;
     }
 
-    public void setContext(Map<String, Object> context) {
+    public void setContext(Context context) {
+        context.validateContextObject();
```

Comment:
Yes I checked the behavior.

This exception gets thrown during Jackson deserialization. This way, a Jackson typical bad request response gets returned, when trying to invalid context object, along with the message, I set in the methods.

Reply:
We can explicitly add a throw clause.
Will add a null check for platform.

Reply:
> I got the following response, when trying to create an instance or update a context with no platform field set. This response is not helpful or appropriate in my opinion.
> 
> ```json
> {
>     "timestamp": "2019-10-01T08:28:21.809+0000",
>     "status": 400,
>     "error": "Bad Request",
>     "message": "JSON parse error: null; nested exception is com.fasterxml.jackson.databind.JsonMappingException: N/A\n at [Source: (PushbackInputStream); line: 20, column: 2] (through reference chain: de.evoila.cf.broker.model.ServiceInstanceUpdateRequest[\"context\"])",
>     "path": "/v2/service_instances/3dd276f2-c7cd-dbeb-0bc6-c506b562f2dc"
> }
> ```
> 
> And does the exception have to be unchecked or can there be a throws clause?

Is this already fixed? I can still recreate the error.

Reply:
This is the standard message returns when a value is missing that it needs to call the setter.
To avoid this, we can validate the value of platform else where, like the getter, and handle validation there.
What do you think @marberger?

Reply:
The error message got changed to a more fitting one. Resolving the comment.